src/pipert2/core/base/flow.py

```diff
@@ -89,10 +89,18 @@ def execute_event(self, event: Method) -> None:
 
         """
 
-        for routine in self.routines.values():
-            routine.execute_event(event)
+        if event.is_valid_by_flow(self.name):
+            base_execution_event = event.create_base_method()
+            routine_name = event.get_routine_name()
```

Comment:
What do you say about adding a method for testing if a the Method has a routine in it instead of taking it immediately.
And then requesting if it exists ?

(It seems a bit petty but i want to make sure is readable as possible)